cookie_source.go

```diff
@@ -11,63 +11,68 @@ import (
 	"golang.org/x/crypto/hkdf"
 )
 
-// CookieSource is used to create and verify source address tokens
+// CookieSource is used to create and verify a cookie
 type CookieSource interface {
 	// NewToken creates a new token
 	NewToken([]byte) ([]byte, error)
 	// DecodeToken decodes a token
 	DecodeToken([]byte) ([]byte, error)
 }
 
-type defaultCookieSource struct {
-	aead cipher.AEAD
+const cookieSecretSize = 32
+const cookieNonceSize = 32
+
+// The DefaultCookieSource is a simple implementation for the CookieSource.
+type DefaultCookieSource struct {
+	secret []byte
 }
 
-const tokenKeySize = 16
-const tokenNonceSize = 16
+var _ CookieSource = &DefaultCookieSource{}
```

Comment:
I am probably not enough of a Golang wizard, but why is this neded.

Reply:
Strictly speaking this is not needed. It makes the compiler check that `DefaultCookieSource` actually implements the `CookieSource` interface. Since it's an assignment to `_`, it will be removed during compilation.